storm-server/src/test/java/org/apache/storm/scheduler/resource/strategies/scheduling/TestLargeCluster.java

```diff
@@ -0,0 +1,458 @@
+     * in matched pairs.
+     *
+     * @throws Exception showing bad and unmatched resource files.
+     */
+    @Test
+    public void testReadSerializedTopologiesAndConfigs() throws Exception {
+        List<String> resources = getResourceFiles(TEST_RESOURCE_PATH);
+        Assert.assertTrue("No resource files found in " + TEST_RESOURCE_PATH, !resources.isEmpty());
+        TopologyDetails[] topoDetailsArray = createTopoDetailsArray(true);
+    }
+
+    /**
+     * Create one supervisor and add to the supervisors list.
+     *
+     * @param rack rack-number
+     * @param superInRack supervisor number in the rack
+     * @param cpu percentage
+     * @param mem in megabytes
+     * @param numPorts number of ports on this supervisor
+     * @param sups returned map os supervisors
+     */
+    private static void createAndAddOneSupervisor(
+            int rack, int superInRack, double cpu, double mem, int numPorts,
+            Map<String, SupervisorDetails> sups) {
+
+        List<Number> ports = new LinkedList<>();
+        for (int p = 0; p < numPorts; p++) {
+            ports.add(p);
+        }
+        String superId = String.format("r%03ds%03d", rack, superInRack);
+        String hostId  = String.format("host-%03d-rack-%03d", superInRack, rack);
+        Map<String, Double> resourceMap = new HashMap<>();
+        resourceMap.put(Config.SUPERVISOR_CPU_CAPACITY, cpu);
+        resourceMap.put(Config.SUPERVISOR_MEMORY_CAPACITY_MB, mem);
+        resourceMap.put("network.resource.units", 50.0);
+        SupervisorDetails sup = new SupervisorDetails(superId,
+                hostId, null, ports,
+                NormalizedResources.RESOURCE_NAME_NORMALIZER.normalizedResourceMap(resourceMap));
+        sups.put(sup.getId(), sup);
+    }
+
+    /**
+     * Create supervisors.
+     *
+     * @param uniformSupervisors true if all supervisors are of the same size, false otherwise.
+     * @return supervisor details indexed by id
+     */
+    private static Map<String, SupervisorDetails> createSupervisors(boolean uniformSupervisors) {
+        Map<String, SupervisorDetails> retVal;
+        if (uniformSupervisors) {
+            int numRacks = 16;
+            int numSupersPerRack = 82;
+            int numPorts = 50; // scheduling will fill up 2-6, leaving of 90% workerslots unused - does this cause slow scheduling?
+            int rackStart = 0;
+            int superInRackStart = 1;
+            double cpu = 7200; // %percent
+            double mem = 356_000; // MB
+            Map<String, Double> miscResources = new HashMap<>();
+            miscResources.put("network.resource.units", 100.0);
+
+            return TestUtilsForResourceAwareScheduler.genSupervisorsWithRacks(
+                    numRacks, numSupersPerRack, numPorts, rackStart, superInRackStart, cpu, mem, miscResources);
+
+        } else {
+            // this non-uniform supervisor distribution closelt (but not exactly) mimics IridiumBlue cluster
```

Comment:
`IridiumBlue` doesn't mean anything in the community code. Maybe reword it.

Typo: `closelt`

Reply:
changed